contracts/tec/contracts/src/mixins/MSignatureValidator.sol

```diff
@@ -0,0 +1,39 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.2;
+
+
+contract MSignatureValidator {
+
+    // Allowed signature types.
+    enum SignatureType {
+        Illegal,         // 0x00, default value
+        EIP712,          // 0x01
+        EthSign,         // 0x02
+        NSignatureTypes  // 0x03, number of signature types. Always leave at end.
```

Comment:
Why "always leave at end"?

Reply:
Kinda just a sanity check to prevent human error when writing this. It's never used, but we have this in the main Exchange contract as well.